### PR TITLE
TS-46 FEATURE - Duplicate Teamsheet

### PR DESCRIPTION
**Pull request by Mark-Ball**
- 'duplicate' and 'create your own' buttons
Screenshot 2024-01-26 at 10 33 09 am

### CODE REVIEW

File: src/app/shared/components/ShareBar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import Button from '@components/Button'
+
+type Router = ReturnType<typeof useRouter>
+
+type CreateAndRedirectParams = {
+  teamSheet: TeamSheet,
+  router: Router,
+}
+
+const createTeamSheetAndRedirect = async (params: CreateAndRedirectParams) => {
```
Comment (dcarg) [inline src/app/shared/components/ShareBar/index.tsx:19]: Think we could just call this duplicateTeamSheet

**Comment by dcarg** [comment]
Sick Juan. All pretty straightforward.

I was thinking when duplicating a teamSheet, would be good to set isEditing to true when redirected to the new teamsheet page so that we are drawing the users attention to naming their new duplicated teamsheet

My first thought was doing it through urlParams but could be a much simpler way of doing it with just a useEffect in the nameForm component. I didn't quite figure it out before I left for the day

Was thinking you pass isTeamSheetComplete param through to NameForm and then run a useEffect checking if a teamsheet is complete and whether it has a title or not or something... thoughts?

```  
useEffect(() => {
    if(!title && !isEditing && isTeamSheetComplete){
      setIsEditing(true)
    }
  }, [id, isEditing, isTeamSheetComplete])
  ```

**Comment by Mark-Ball** [comment]
good idea, the useEffect i put in there now opens the input field if the teamsheet is complete and there is no title, so once the user completes their teamsheet this field will open

**Review by dcarg — APPROVED** [review]
Yep nice!

**Pull request merged**